### PR TITLE
Improve Firmware Docs

### PR DESCRIPTION
**Pull request by SiriusStarr**
First stab at documenting how to integrate custom/userspace firmware with the sval stuff.  This is aimed at a pretty QMK-/C-aware crowd, since no one else should need to do this.

(@morganvenable Do you want lots of independent PRs or one big one?)

### CODE REVIEW

**Comment by morganvenable** [comment]
Fewer is better.  Ultimately docs are read as-is, noy super interested in
more clicks

**Comment by SiriusStarr** [comment]
Aight, let's leave this unmerged until I'm done then and I'll make it a general docs PR.

File: keyboards/svalboard/docs/images/trackball.jpg
Comment (SiriusStarr) [inline keyboards/svalboard/docs/images/trackball.jpg:1]: @morganvenable If you have better images of the trackball/trackpoint, send them to me; I just ripped these from the shop.

File: keyboards/svalboard/docs/images/trackpoint.jpg
Comment (SiriusStarr) [inline keyboards/svalboard/docs/images/trackpoint.jpg:1]: @morganvenable If you have better images of the trackball/trackpoint, send them to me; I just ripped these from the shop.

**Comment by SiriusStarr** [comment]
I think we're good on this as far as strict firmware things go, if you want to review and merge.

I'll do a second PR with some more "subjective" things:

* Basic "How2Vial"
* Some very basic troubleshooting (dark thumb being the big thing, also not flashing both sides, wrong trackball, etc)
* Some very basic, unopinionated layout design stuff (not advocating against QWERTY or anything, just more of a "here are layout things you can do that you might not know about if you're new to this world" [HRMs, combos, overrides, etc.] and how to do those in Vial).

A third PR to clean up/document the *.vils.

And a fourth PR to rip out the root level docs and tie them all together nicely.

(Unless you feel strongly about keeping them all together in one PR.)


(Note that these docs assume we merge the Caps Word fix in #64 and shift invert in #65, but those should be non-controversial changes, even if the implementation needs fixing.)

**Comment by morganvenable** [comment]
No strong feelings.  approving momentarily.  Thank you so much for all of
this.

**Review by morganvenable — APPROVED** [review]
Lgtm

**Comment by SiriusStarr** [comment]
Saying it here, since this is kind of my living document for doc improvements, but I'd like to clean up and document the *.vils we provide too.

**Comment by morganvenable** [comment]
god yes.  I just don't want to put any more of my stupid misordered stuff
out there...

**Comment by SiriusStarr** [comment]
Since they're just JSON, they actually version control pretty well, so it should be maintainable.

**Comment by ilc** [comment]
LGTM,

**Pull request merged**